src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -0,0 +1,48 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+//Servlet to authenticate users before they create or join a Room
+@WebServlet("/auth")
+public class AuthServlet extends HttpServlet{
+    private static final String REDIRECT_URL_PARAMETER = "redirectUrl";
+    private static final String URL_TO_REDIRECT_AFTER_LOG_OUT = "/";
+    
+    @Override
+    public void doGet(HttpServletRequest req, HttpServletResponse res) throws IOException{
+
+        Boolean isUserLoggedIn = isUserLoggedIn(ServletUtil.USER_SERVICE);
+        String urlToRedirectToAfterUserLogsIn = req.getParameter(REDIRECT_URL_PARAMETER);
+        String responseBody = getResponseBody(isUserLoggedIn, urlToRedirectToAfterUserLogsIn, ServletUtil.USER_SERVICE);
+        if(isUserLoggedIn){
+            res.setContentType(ServletUtil.JSON_CONTENT_TYPE);
+            res.getWriter().println(ServletUtil.PARSER.toJson(responseBody));  
+        }
+        else{
+            res.setContentType(ServletUtil.HTML_CONTENT_TYPE);
+            res.getWriter().println(responseBody);
+        }
+    }
+
+    public static Boolean isUserLoggedIn(UserService userService){
```

Comment:
This function seems useless. Its callers can call the isUserLoggedIn directly instead of calling this static function.

Reply:
This was to help with testing, Killian asked for the function to be what is actually used in the servlet, so I had to refactor it.

Reply:
To clarify, you shouldn't duplicate logic in the test file for testing, but rather call the Servlet logic directly. Creating separate functions isn't necessary unless it breaks up the code. In this case you can call the doGet which is what I think Tianxiao wants.

Reply:
+1 with what Killian said.

Reply:
This has been fixed